api/api.go

```diff
@@ -25,6 +26,60 @@ type PublicController struct {
 	Database *sql.DB
 }
 
+func (f *PublicController) GetIdFromCookie(c *gin.Context) int {
+	Cookie := c.Request.Header.Get("Cookie")
+	if Cookie == "" {
+		log.Errorf("Unable to get cookie")
+		c.IndentedJSON(http.StatusUnauthorized, "Unable to get cookie")
+		return -1
+	}
+	tokenString := strings.Split(Cookie, "=")[1]
+
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		return []byte("foobar"), nil
```

Comment:
Nicht sicher, bitte nach dem Beispiel in den Docs abändern. Es sollte ein byte slice mit dem Secret zurückgegeben werden.
https://pkg.go.dev/github.com/dgrijalva/jwt-go/v4#example-Parse-Hmac

Reply:
das  Secret ist der Key mit dem das token gebaut wurde in dem Fall "foobar" können wir aber noch ändern wenn das lieber ne globale variable oder so sein soll aber prinzipiell entspricht das genau dem Doc das du geschickt hast

Reply:
Inwiefern? Die hard coden das dort nicht und geben nicht direkt an woher das kommt.

Das Secret ist schon in der `config.toml` drinnen und wurde vorher benutzt, du hast es aber rausgenommen:
https://github.com/LearningBay24/LearningBay24-backend/pull/61/files#diff-bf98d5fab5bcfabded9769069e7318683f549ec91cff84c5f87a9ae3a3ca3d8dL265